CommonTools/ParticleFlow/src/PFPileUpAlgo.cc

```diff
@@ -22,7 +22,9 @@ void PFPileUpAlgo::process(const PFCollection& pfCandidates, const reco::VertexC
 
     // no associated vertex, or primary vertex
     // not pile-up
-    if (ivertex == -1 || ivertex == 0) {
+    if ((ivertex == -1 || ivertex == 0 ||
+         (fNumOfPUVtxsForCharged_ > 0 && !vertices.empty() && ivertex <= int(fNumOfPUVtxsForCharged_) &&
+          std::abs(cand.vertex().z() - vertices.at(0).z()) < fDzCutForChargedFromPUVtxs_))) {
```

Comment:
Typically for performance reasons the non-throwing `vertices[0]` would be preferred. It looks like it's guaranteed by `!vertices.empty()` that there is at least one vertex.